helix-term/src/ui/editor.rs

```diff
@@ -110,26 +110,26 @@ impl EditorView {
                     .and_then(|source| source.path.as_ref())
                     == doc.path()
             {
-                let line = frame.line - 1; // convert to 0-indexing
-                let style = theme.get("ui.highlight");
+                let line = frame.line;
+                let style = theme.get("ui.highlight.currentline");
                 let line_decoration = move |renderer: &mut TextRenderer, pos: LinePos| {
                     if pos.doc_line != line {
                         return;
                     }
                     renderer
                         .surface
-                        .set_style(Rect::new(area.x, pos.visual_line, area.width, 1), style);
+                        .set_style(Rect::new(inner.x, pos.visual_line, inner.width, 1), style);
                 };
 
                 line_decorations.push(Box::new(line_decoration));
             }
         }
 
-        if is_focused && config.cursorline {
+        if stack_frame.is_none() && is_focused && config.cursorline {
             line_decorations.push(Self::cursorline_decorator(doc, view, theme))
         }
 
-        if is_focused && config.cursorcolumn {
+        if stack_frame.is_none() && is_focused && config.cursorcolumn {
             Self::highlight_cursorcolumn(doc, view, surface, theme, inner, &text_annotations);
         }
```

Comment:
Didn't make much sense too me too keep them, at least cursorline, since you basically get one with a different style with this change. I was also afraid they would overrule the debug line

Reply:
With the changes you made to the code, the cursorline and cursorcolumn are always disabled when a frame is active so also if the cursor is anywhere else (so not just on the line with the breakpoint).

To make sure that the cursorline doesn't overwrite the frame highlight, simply mode the adsition of the fram highlight after the cursor line and cursorcolumn code. The last line decoration always overwrites previous ones.

Reply:
That's what I initially did, then thought it was too easy :sweat,_smile: will fix tomorrow